### ISSUE TITLE
When watching for new blocks on local `geth` instance, callback isn't fired after block 255

### ISSUE THREAD

**Issue by ahoppen** [MetaMask/metamask-extension#6423]
**Describe the bug**
When instantiating a local `geth` blockchain and watching for new blocks using `web3.eth.filter('latest', callback)` the callback isn't fired after block 255 is mined. I'm not 100% sure if this is a MetaMask or `geth` issue but am lacking the knowledge to investigate further.

**To Reproduce**
1. Launch `geth` using `geth --dev --datadir /path/to/empty/dir/ --dev.period 1 --rpc --rpccorsdomain "*" console`
2. Unlock MetaMask and point it to `Localhost 8545`
3. On any webpage open the console and execute
```javascript
web3 = new Web3(ethereum);
web3.eth.filter('latest', (error, result) => {if (error) {console.error(error);} else { console.log(result);}});
```
4. Watch the following: As a new block is mined every second, it shows up in the terminal window in which you launched `geth`. Shortly afterwards, the block hash shows up in the browser's window
5. Wait 255 seconds until the 255th block is mined.

**Expected behavior**
After the 255th block, new block hashes should continue to show up in the browser's console.

**Actual behaviour**
The block hashes only appear up until the 255th block. After that, the callback isn't fired anymore.

**Reproduction screen recording**
You can find a video of a reproduction here: https://www.dropbox.com/s/3dck61fsz9hvrr5/Reproduction%20video.mp4?dl=0

In there you can see:
0:00 – 0:04: Starting the local geth node
0:04 – 0:18: Pointing MetaMask to the local geth instance
0:18 – 0:28: Setting up watching for new blocks
0:28 – 4:18: Waiting for the first 255 blocks to be mined (really boring)
4:20: Callbacks stop firing
4:41: The last block for which the callback was fired is block 255
4:46 – 5:10: Showing versions of components used

**Browser details**
 - OS: macOS 10.14.3
 - Browser: Chrome 73.0.3683.86
 - MetaMask Version: 6.3.1
 - `geth` version: 1.8.22-stable

**Comment by ahoppen** [comment]
Is there anything I can do to help reproduce/fix this issue?

**Comment by tmashuang** [comment]
Fixed in https://github.com/MetaMask/eth-json-rpc-filters/pull/7 will be fixed in the next version of MetaMask. Thanks for the reproduction, and sorry we missed this!

**Comment by ahoppen** [comment]
Thanks. I'll verify when the next version of MetaMask is released.

**Comment by ahoppen** [comment]
Verified that this has been fixed in MetaMask 6.5.3. 

Thanks a lot!
